apps/climatemappedafrica/src/lib/hurumap/index.js

```diff
@@ -160,3 +163,96 @@ export async function fetchProfileGeography(
     items: formatProfileGeographyData(data, parent),
   };
 }
+
+async function getProfileCachedData(cacheFile, profileId) {
+  try {
+    const cachedData = await fs.readFile(cacheFile, "utf-8");
+    const { data, timestamp } = JSON.parse(cachedData);
+    return { data, timestamp };
+  } catch (error) {
+    console.error(`No valid cache found for profile ${profileId}`);
+    return null;
+  }
+}
+
+export async function fetchCachedProfile(params, maxCacheAge = 10) {
+  const { profileId } = params;
+  const cacheDir = path.resolve(process.cwd(), "public", "cache");
+  const cacheFile = path.join(cacheDir, `profile-${profileId}.json`);
+
+  try {
+    await fs.mkdir(cacheDir, { recursive: true });
+
+    const cached = await getProfileCachedData(cacheFile, profileId);
+    if (cached) {
+      const cacheAge = Date.now() - cached.timestamp;
+      const maxCacheAgeInMs = maxCacheAge * 60 * 1000;
+
+      if (cacheAge < maxCacheAgeInMs) {
+        return cached.data;
+      }
+    }
+
+    const profileData = await fetchProfile(params);
+    const cacheContent = JSON.stringify({
+      timestamp: Date.now(),
+      data: profileData,
+    });
+
+    await fs.writeFile(cacheFile, cacheContent, "utf-8");
```

Comment:
Minor, Issue but since I/O is expensive could we check if the newly fetched data has really changed before we write to the file ? (Maybe compare hashes) 🤔 , maybe something that could be worth considering is how to handle race conditions, It could be I'm worried about something very improbable, but looking at the code there is a chance of the cache being corrupted